frontend/src/hooks/useUserSession.js

```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
+
+import { useExceptionHandler } from "../hooks/useExceptionHandler.jsx";
+import { useAlertStore } from "../store/alert-store";
+const useUserSession = () => {
+  const handleException = useExceptionHandler();
+  const { setAlertDetails } = useAlertStore();
+  return async () => {
+    try {
+      const requestOptions = {
+        method: "GET",
+        url: "/api/v1/session",
+      };
+      const res = await axios(requestOptions);
+      return res.data;
+    } catch (error) {
+      if (error?.response?.statusText === "Payment Required") {
+        handleException(error, "Error while getting session");
+      } else if (error?.response?.statusText === "Unauthorized") {
+        handleException(error, "Error while getting session");
```

Comment:
@athul-rs Can we store this fallback error message in a variable and use that in all three places?

Reply:
@tahierhussain Sure.